include/psa/crypto_extra.h

```diff
@@ -1901,6 +1901,342 @@ static inline struct psa_pake_operation_s psa_pake_operation_init( void )
+ *
+ */
+struct psa_sign_interruptible_context_s
+{
+    size_t MBEDTLS_PRIVATE(num_ops);
+};
+
+typedef struct psa_sign_interruptible_context_s psa_sign_interruptible_context;
+
+/**
+ * \brief                       Set the maximum number of basic operations done
+ *                              in a row.
+ *
+ * \note                        Although the interpretation of this value is
+ *                              implicitly implementation defined, this value
+ *                              needed a default, thus this value defaults to
+ *                              1, which means the absolute minimum of work
+ *                              will be done each interruptible call.
+ *
+ * \note                        If more operations are needed to complete a
+ *                              computation, #PSA_ERROR_IN_PROGRESS will be
+ *                              returned by the function performing the
+ *                              computation. It is then the caller's
+ *                              responsibility to either call again with the
+ *                              same parameters until it returns 0 or an error
+ *                              code; or to free the relevant context if the
+ *                              operation is to be aborted.
+ *
+ * \note                        It is strictly required that all input
+ *                              parameters and the restart context be the same
+ *                              on successive calls for the same operation,
+ *                              but output parameters need not be the same;
+ *                              they must not be used until the function
+ *                              finally returns 0.
+ *
+ *                              This only applies to functions whose
+ *                              documentation mentions they may return
+ *                              #PSA_ERROR_IN_PROGRESS.
+ *
+ * \param max_ops               The maximum number of basic operations to
+ *                              be done in a row. Passing zero here has the
+ *                              same effect as passing the maximum value - i.e
+ *                              the operation will be done in one go. Lower
+ *                              (non-zero) values mean signature functions
+ *                              will block for a lesser maximum amount of time
+ *                              and conversely larger values will mean
+ *                              blocking for a larger maximum amount of time.
+ *                              The effect of changing this value will depend
+ *                              on the key used, the curve used and the
+ *                              underlying implementation, so we provide /c
+ *                              psa_sign_interruptible_get_num_ops() in order
+ *                              to help with tuning this value.
+ *
+ */
+void psa_sign_interruptible_set_max_ops( size_t max_ops );
```

Comment:
I'll keep my comment above so others can see my reasoning, but [Manuel's comment](https://github.com/Mbed-TLS/mbedtls/pull/6279#issuecomment-1260747896) made me change my mind.

> On the other hand, when the operation happens via another layer (X.509, TLS), this layer is now responsible for deciding what max_ops to pass. We could provide things like mbedtls_ssl_conf_set_interruptible_max_ops() but we run again into the problem that we don't know what "basic operation" means until the curve (or other) has been chosen, which will happen as part of the handshake - so we haven't solved the problem just moved it to other layers.

And since indeed

> "how much time is it acceptable to block" does not depend on the operation but on the system

it's pointless to make the intermediate layer responsible for picking `max_ops`. So a global setting makes more sense.

Mind you, we could even have a global default plus a per-operation setting. It doesn't seem hard to implement — again, we only care about single-threaded systems, so the extra cost is just a little bit more code size and RAM usage. Is even that little cost worth the added flexibility though?